controllers/lookup/k8s_client.go

```diff
@@ -0,0 +1,160 @@
+package lookup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	internalhttp "github.com/datainfrahq/druid-operator/pkg/http"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type K8sClient struct {
+	client client.Client
+}
+
+func NewK8sClient(client client.Client) K8sClient {
+	return K8sClient{
+		client: client,
+	}
+}
+
+func (c *K8sClient) FindLookups(ctx context.Context, reports map[types.NamespacedName]Report) (map[ClusterKey]map[LookupKey]Spec, error) {
+	specs := &v1alpha1.DruidLookupList{}
+	if err := c.client.List(ctx, specs); err != nil {
+		return nil, err
+	}
+
+	lookupSpecsPerCluster := make(map[ClusterKey]map[LookupKey]Spec)
+	for _, spec := range specs.Items {
+		clusterKey := ClusterKey{
+			Namespace: spec.Namespace,
+			Cluster:   spec.Spec.DruidClusterName,
+		}
+		lookupKey := LookupKey{
+			Tier: spec.Spec.Tier,
+			Id:   spec.Spec.Id,
+		}
+		var lookupSpec interface{}
+		if err := json.Unmarshal([]byte(spec.Spec.Spec), &lookupSpec); err != nil {
+			setIfNotPresent(
+				reports,
+				spec.GetNamespacedName(),
+				Report(NewErrorReport(
+					fmt.Errorf(
+						"lookup resource %v in cluster %v/%v contains invalid spec, should be JSON",
+						spec.Name,
+						clusterKey.Namespace,
+						clusterKey.Cluster,
+					),
+				)),
+			)
+			continue
+		}
+
+		if lookupKey.Tier == "" {
```

Comment:
Can be removed and replaced with kubebuilder:default marker: https://book.kubebuilder.io/reference/markers/crd-validation#:~:text=//%20%2B-,kubebuilder%3Adefault,-any

Reply:
Addressed in 70b96b6 and b0ce7ff.